tests/providers/gcp/e2e_cli.py

```diff
@@ -0,0 +1,275 @@
+          snapshot of disk passed to the 'disk_name' parameter in the
+          forensics.CreateDiskCopy() method.
+      2. A new disk is created from the taken snapshot.
+      3. If an analysis VM already exists, the module will attach the disk
+          copy to the VM. Otherwise, it will create a new GCP instance for
+          analysis purpose and attach the boot disk copy to it.
+    """
+
+    # Make a copy of another disk of the instance to analyse
+    self.disk_to_forensic_copy = self.cli.CreateDiskCopy(
+        disk_name=self.disk_to_forensic)
+
+    gcp_client_api = common.GoogleCloudComputeClient(self.project_id).GceApi()
+
+    # The disk copy should be existing in the analysis project
+    gce_disk_client = gcp_client_api.disks()
+    request = gce_disk_client.get(
+        project=self.project_id, zone=self.zone,
+        disk=self.disk_to_forensic_copy.name)
+    result = request.execute()
+    self.assertEqual(result['name'], self.disk_to_forensic_copy.name)
+
+    # Get the analysis VM and attach the evidence disk to forensic
+    self.analysis_vm = self.cli.StartAnalysisVm(
+        self.instance_to_analyse,
+        self.zone,
+        attach_disks=[self.disk_to_forensic_copy.name])
+
+    # The forensic instance should be live in the analysis GCP project and
+    # the disk should be attached
+    gce_instance_client = gcp_client_api.instances()
+    request = gce_instance_client.get(
+        project=self.project_id, zone=self.zone, instance=self.analysis_vm_name)
+    result = request.execute()
+    self.assertEqual(result['name'], self.analysis_vm_name)
+
+    for disk in result['disks']:
+      if disk['boot']:
+        request = gce_disk_client.get(project=self.project_id,
+                                      zone=self.zone,
+                                      disk=disk['source'].split('/')[-1])
+        boot_disk = request.execute()
+        self.assertEqual('pd-ssd', boot_disk['type'].rsplit('/', 1)[-1])
+      if disk['source'].split('/')[-1] == self.disk_to_forensic_copy.name:
+        return
+    self.fail(
+        'Error: could not find the disk {0:s} in instance {1:s}'.format(
+            self.disk_to_forensic_copy.name, self.analysis_vm_name))
+
+  @classmethod
+  @typing.no_type_check
+  @IgnoreWarnings
+  def tearDownClass(cls):
+    analysis_vm = cls.analysis_vm
+    zone = cls.zone
+    project = cls.gcp
+    disks = analysis_vm.ListDisks()
+    gcp_client = common.GoogleCloudComputeClient(project.project_id)
+    # delete the created forensics VMs
+    logger.info('Deleting analysis instance: {0:s}.'.format(
+        analysis_vm.name))
+    gce_instance_client = gcp_client.GceApi().instances()
+    request = gce_instance_client.delete(
+        project=project.project_id, zone=zone, instance=analysis_vm.name)
+    response = request.execute()
+    try:
+      gcp_client.BlockOperation(response)
+    except HttpError:
+      # BlockOperation triggers a while(True) loop that checks on the
```

Comment:
Should we have a bug/issue open that tracks this and changes it to a tighter While loop (on a condition that changes when the operation has finished?)

Reply:
Agreed, I'll open one, thanks!